### PR TITLE
trigger an error if the api script is not loaded

### PR DESCRIPTION
**Pull request by andrepolischuk**
also I rewrited mounting the api script with global `Promise` to fix case with multiple captchas. if multiple captchas are using, handleOnLoad callback is pushed to the onLoadListeners only for first component. next captchas in componentDidMount have apiScriptRequested true, so handleOnLoad is not adding to callback's array

### CODE REVIEW

**Comment by andrepolischuk** [comment]
@zoryana94 @DSergiu hi, could you take a look at this pr?

**Comment by andrepolischuk** [comment]
@zoryana94 @DSergiu ping

File: src/index.js [hunk 1]
```diff
@@ -186,6 +195,12 @@ class HCaptcha extends React.Component {
       });
     }
 
+    handleOnLoadingError (event) {
+      const { onError } = this.props;
+
+      if (onError) onError(event);
+    }
+
```
Comment (brdlyptrs) [inline src/index.js:203]: Let's utilize the `handleError` below, we can update it like so:

```js
 handleError (event) {
      const { onError } = this.props;
      const { captchaId } = this.state;

      if (this.isReady())
           // If hCaptcha runs into error, reset captcha - hCaptcha
           hcaptcha.reset(captchaId)

      if (onError) onError(event);
    }
```

File: src/index.js [hunk 2]
```diff
@@ -1,34 +1,45 @@
...
   delete params.apihost;
 
   const script = document.createElement("script");
-  script.src = `${domain}/1/api.js?render=explicit&onload=hcaptchaOnLoad`;
+  script.id = SCRIPT_ID;
+  script.src = `${domain}/1/api.js?render=explicit&onload=${HCAPTCHA_LOAD_FN_NAME}`;
   script.async = true;
+  script.onerror = (event) => {
+    console.error('Failed to load api: ' + script.src, event);
+    rejectFn(new Error('Failed to load api'));
```
Comment (brdlyptrs) [inline src/index.js:34]: Let's use the [pattern](https://docs.hcaptcha.com/configuration#error-codes) we have for other errors we propagate to the user: `script-error`

**Comment by brdlyptrs** [comment]
Hi @andrepolischuk thanks for the PR. Looks pretty good overall, I've added a few comments above.

File: src/index.js (hunk 2 above)
Comment (andrepolischuk) [reply src/index.js]: +

File: src/index.js (hunk 1 above)
Comment (andrepolischuk) [reply src/index.js]: good idea, thanks. refactored

File: src/index.js [hunk 3]
```diff
@@ -1,34 +1,45 @@
...
   const domain = params.apihost || "https://js.hcaptcha.com";
   delete params.apihost;
 
   const script = document.createElement("script");
-  script.src = `${domain}/1/api.js?render=explicit&onload=hcaptchaOnLoad`;
+  script.id = SCRIPT_ID;
+  script.src = `${domain}/1/api.js?render=explicit&onload=${HCAPTCHA_LOAD_FN_NAME}`;
   script.async = true;
+  script.onerror = (event) => {
+    console.error('Failed to load api: ' + script.src, event);
```
Comment (brdlyptrs) [inline src/index.js:33]: Just noticed this, we should probably remove this as the error can be logged by the user if they wish to do so.

**Comment by brdlyptrs** [comment]
One last note, looks good! 

Just to confirm, as it looks like it, this update will handle multiple hCaptcha components loading at the same time waiting for the script tag to load correct?

File: src/index.js (hunk 3 above)
Comment (andrepolischuk) [reply src/index.js]: done

**Comment by andrepolischuk** [comment]
> Just to confirm, as it looks like it, this update will handle multiple hCaptcha components loading at the same time waiting for the script tag to load correct?

yes. first instance of the component mounts the script and waits promise (in module root) to handle onload. the rest of instances check mounted script in dom, and wait the same promise to handle onload.

**Comment by brdlyptrs** [comment]
I think we can bump this version to `1.3.1` and then merge.

**Comment by andrepolischuk** [comment]
@brdlyptrs pull latest master into this branch

**Comment by andrepolischuk** [comment]
@brdlyptrs are you planning to publish this soon?

**Review by brdlyptrs — APPROVED** [review]
We can version bump separately.

**Review by zoryana94 — APPROVED** [review]

**Pull request merged**